### DISCUSSION TITLE
Reducing the output of the Maven plugin CRD generator

### DISCUSSION THREAD

**Discussion by heruan** [fabric8io/kubernetes-client#5814] [Q&A]
I'm using the Maven plugin CRD generator and it works well, but generates a lot of output, i.e.

```
[INFO] Compiling 1178 source files to /home/runner/work/control-center/control-center/control-center-crds/target/classes
  0% Generating: org.keycloak.k8s.v2alpha1.keycloakspec.unsupported.podtemplate.spec.volumes.downwardapi.items.ResourceFieldRef
  0% Generating: org.keycloak.k8s.v2alpha1.keycloakspec.unsupported.podtemplate.spec.containers.envfrom.ConfigMapRef
  0% Generating: io.cnpg.postgresql.v1.clusterspec.Affinity
  0% Generating: io.cert_manager.acme.v1.challengespec.solver.http01.ingress.podtemplate.spec.affinity.nodeaffinity.PreferredDuringSchedulingIgnoredDuringExecution
  0% Generating: org.keycloak.k8s.v2alpha1.keycloakspec.unsupported.podtemplate.spec.initcontainers.securitycontext.SeLinuxOptions
  0% Generating: io.cert_manager.v1.clusterissuerspec.acme.solvers.http01.ingress.podtemplate.spec.affinity.podantiaffinity.RequiredDuringSchedulingIgnoredDuringExecution
  1% Generating: io.cert_manager.acme.v1.challengespec.solver.http01.ingress.podtemplate.spec.affinity.PodAffinity
  1% Generating: io.cert_manager.acme.v1.challengespec.solver.dns01.Digitalocean
  1% Generating: org.keycloak.k8s.v2alpha1.keycloakspec.unsupported.podtemplate.spec.initcontainers.startupprobe.HttpGet
  1% Generating: io.cnpg.postgresql.v1.poolerspec.template.spec.initcontainers.lifecycle.poststart.httpget.HttpHeaders
...
```

The output clutters the logs in GitHub actions and I guess it also slows down execution with a lot of I/O. I tried to silence it with `mvn -q` but looks like the `-q` option is ignored by the plugin (and I don't want to silence everything, just the 0%-100% progress output).

How can I do that?

**Comment by andreaTP** [comment]
Hi @heruan ! And thanks for your interest in the project.
I believe that the output is from the `sundrio` generation:
https://github.com/sundrio/sundrio/blob/3d6d36b75c90061f1a7fc700af0fc9caf58a1643/annotations/builder/src/main/java/io/sundr/builder/internal/processor/AbstractBuilderProcessor.java#L218

apparently, this is logged to stderr so something like `2> /dev/null` should silence that output, would this work for you?

**Reply by heruan** [reply]
Thanks! I cannot silence all STDERR otherwise the CI/CD workflow could miss real errors. I reported this in https://github.com/sundrio/sundrio/issues/464 also.
